src/System.Device.Gpio/System/Device/I2c/I2cCommunicationException.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace System.Device.Gpio.I2c
+{
+    /// <summary>
+    /// This exception is thrown on I2C communication errors
+    /// </summary>
+    public class I2cCommunicationException : IOException
```

Comment:
Currently, we keep getting very unspecific exceptions from I2C problems, typically "Error 121", but also TimeoutExceptions and the like. So I thought it would be good to have something more specific (but then use it for all kinds of errors in I2C communication). But I have no strong oppinion on this one.

Reply:
I originally thought like you and with the first implementation I realize we never really care about the details of the exception except for debug. Rest of the time, you just want the device to work and work and always work. So worst case scenario, in the main code, you trap all Exceptions by category with retry methods so your code never break. And tracking the IOException is more than largely enough.